src/states/common/base.rs

```diff
@@ -304,12 +316,50 @@ pub trait Base: Display {
         &mut self,
         addr: SocketAddr,
         msg: Bytes,
-        token: Token,
+        msg_token: Token,
         _outbox: &mut dyn EventBox,
     ) -> Transition {
-        let log_ident = LogIdent::new(self);
-        self.network_service_mut()
-            .send_message_to_next_target(msg, token, addr, log_ident);
+        let next_target = self.network_service_mut().target_failed(msg_token, addr);
+        match next_target.failed_attempts {
+            0 => {
+                trace!(
+                    "{} - Sending message ID {} to {} failed - resending to {} now",
+                    self,
+                    msg_token,
+                    addr,
+                    next_target.addr
+                );
+                self.network_service_mut()
+                    .send_now(next_target.addr, msg, msg_token);
+            }
+            n if n < RESEND_MAX_ATTEMPTS => {
```

Comment:
Commit: Delay resending to previously failed target 

Wonder if it would make sense to keep most of it where it was, i.e out of base. and maybe use the result to trigger a scheduling for example something like:

```rust
if let Some(shedule) = self.network_service_mut().send_message_to_next_target(msg, token, addr, log_ident) {
    let timer_token = self.timer().schedule(RESEND_DELAY);
    self.network_service_mut().send_later(timer_token, shedule);
}
```

i.e the RESEND_MAX_ATTEMPTS is still where it was as well as most of the code?

Reply:
Sure. Do you think that would be clearly better than the current way? If so, why?

Reply:
The initial design was trying to keep the resend logic completely separated from the state.
Now it is no longer completely possible because of the timer, but the reason I think it would be better is that States already do a lot, and it seem we should strive to avoid adding code to it when not necessary (i.e push it to the helper code).

Reply:
OK, that makes sense. I'll see what I can do.